src/solver/optimisation/opt_appel_solveur_lineaire.cpp

```diff
@@ -85,24 +85,19 @@ struct SimplexResult
     mpsWriterFactory mps_writer_factory;
 };
 
-static void fillModelerComponents(std::vector<std::unique_ptr<ComponentFiller>>& componentFillers,
-                                  std::vector<LinearProblemFiller*>& fillersCollection,
-                                  const Antares::Study::SystemModel::System* modelerSystem)
+static auto buildComponentFillers(const Antares::Study::SystemModel::System* modelerSystem)
 {
-    if (modelerSystem == nullptr)
+    std::vector<LinearProblemFiller*> componentFillers;
+    if (!modelerSystem)
     {
-        return;
+        return componentFillers;
     }
 
-    for (const auto& [_, component]: modelerSystem->Components())
+    for (const auto& component: modelerSystem->Components() | std::views::values)
     {
-        auto cf = std::make_unique<ComponentFiller>(component);
-        componentFillers.push_back(std::move(cf));
-    }
-    for (auto& component_filler: componentFillers)
-    {
-        fillersCollection.push_back(component_filler.get());
+        componentFillers.push_back(std::make_unique<ComponentFiller>(component).get());
```

Comment:
This is not equivalent to the previous code.

Because you're not keeping the `std::unique_ptr` alive, it will release memory at the end of scope. `componentFillers` will contain dangling pointers after function `buildComponentFillers` returns it.

A strong indication for this is `std::make_unique<Stuff>(args...).get()`, which you might call a code smell.